### ISSUE TITLE
sqlfluff fix adding too many newlines to bench_002_pearson.sql

### ISSUE THREAD

**Issue by WittierDinosaur** [sqlfluff/sqlfluff#1344]
This happens when running tox -e bench, output can be observed in /benchmarks/bench/bench_002_pearson_fix.sql

## Version
Current main, I jumped back about 20 commits, and it still happens

## Configuration
No deviation from standard configs in main branch

**Comment by barrywhart** [comment]
To clarify how to reproduce this, run:
```
sqlfluff fix ./benchmarks/bench_002/bench_002_pearson.sql
```

It adds 9 blank lines between the comment block and the code following it for every column in the first `SELECT` statement, e.g.

Before:
```
    -- The following is the slope of the regression line. Note that CORR (which is the Pearson's correlation
    --  coefficient is symmetric in its arguments, but since STDDEV_POP(open_rate_su) appears in the
    --  numerator this is the slope of the regression line considering STDDEV_POP(open_rate_su) to be
    --  the "y variable" and uses_small_subject_line to be the "x variable" in terms of the regression line.
    ,SAFE_DIVIDE(SAFE_MULTIPLY(CORR(open_rate_su, uses_small_subject_line), STDDEV_POP(open_rate_su)), STDDEV_POP(uses_small_subject_line)) AS open_uses_small_subject_line
```

After:
```
    -- The following is the slope of the regression line. Note that CORR (which is the Pearson's correlation
    --  coefficient is symmetric in its arguments, but since STDDEV_POP(open_rate_su) appears in the
        --  numerator this is the slope of the regression line considering STDDEV_POP(open_rate_su) to be
        ,--  the "y variable" and uses_small_subject_line to be the "x variable" in terms of the regression line.
        








        SAFE_DIVIDE(
             SAFE_MULTIPLY(
                CORR(open_rate_su, uses_small_subject_line),
                STDDEV_POP(open_rate_su)
            ),
             STDDEV_POP(uses_small_subject_line)
        ) AS open_uses_small_subject_line
```

**Comment by barrywhart** [comment]
I don't know if this is a smoking gun, but it should probably be addressed:
```
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  4, P:  1] delete:','>, <LintFix: create @[L:  3, P: 23] create:','>, <LintFix: delete @[L: 10, P:  1] delete:','>, <LintFix: create @[L:  9, P:  1] create:','>]
```

TL;DR: The leading/trailing commas rule should ignore comments. Rather than moving the leading comma before `SAFE_DIVIDE` to follow `business_type`, it moves it to the line _immediately_ above, which is a comment.

**Comment by barrywhart** [comment]
If I apply  _just_ `L019`, the comma before `SAFE_DIVIDE` gets correctly moved to follow `business_type`. Interestingly, though, this requires several iterations -- it moves the comma up one line at a time. This seems like more evidence we need to fix this.

Related thought: I'm guessing that when L003 and L019 are both enabled, L003 somehow interferes with L019, creating (or at least contributing to) the apparently endless looping behavior.

```
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  4, P:  1] delete:','>, <LintFix: create @[L:  3, P: 23] create:','>, <LintFix: delete @[L: 10, P:  1] delete:','>, <LintFix: create @[L:  9, P:  1] create:','>]
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  9, P:  1] delete:','>, <LintFix: create @[L:  8, P:  1] create:','>]
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  8, P:  1] delete:','>, <LintFix: create @[L:  7, P:  1] create:','>]
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  7, P:  1] delete:','>, <LintFix: create @[L:  6, P:  1] create:','>]
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  6, P:  1] delete:','>, <LintFix: create @[L:  5, P:  1] create:','>]
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  5, P:  1] delete:','>, <LintFix: create @[L:  4, P: 15] create:','>]
INFO       Fix loop complete. Stability achieved after 6/10 loops.
```

**Comment by barrywhart** [comment]
I found another problem. The "infinite loop" consists of the following fixes over and over:
```
INFO       [L016] Unfixable inline comment, alone on line: <CommentSegment: ([L:  6, P:  1]) "-- The following is the slope of the regression line. Note that CORR (which is the Pearson's correlation">
INFO       [L016] Unfixable inline comment, alone on line: <CommentSegment: ([L:  7, P:  1]) '--  coefficient is symmetric in its arguments, but since STDDEV_POP(open_rate_su) appears in the'>
INFO       [L016] Unfixable inline comment, alone on line: <CommentSegment: ([L:  8, P:  1]) '--  numerator this is the slope of the regression line considering STDDEV_POP(open_rate_su) to be'>
INFO       [L016] Attempting move of inline comment at end of line: <CommentSegment: ([L:  9, P:  1]) '--  the "y variable" and uses_small_subject_line to be the "x variable" in terms of the regression line.'>
INFO       Applying Fixes [L016]: [<LintFix: delete @[L:  9, P:  1] delete:'--  the "y variable" and uses_small_subject_line to be the "x variable" in terms of the regression line.'>, <LintFix: create @[L:  9, P:  1] create:'    --  the "y variable" and uses_small_subject_line to be the "x variable" in terms of the regression line.\n'>]
INFO       Applying Fixes [L019]: [<LintFix: delete @[L:  9, P:  1] delete:','>, <LintFix: create @[L:  9, P:  1] create:','>]
```

Two observations:
* L019 is deleting and creating a comma at the same location. This is not useful at all.
* L016 ("line too long") is prepending 4 spaces to the beginning of the line and adding a newline to the end. This makes the line _longer_, as well as adding another blank line.

Both of these seem like they _may_ have trivial fixes. The L016 issue looks like the worst of the issues discovered thus far, because it would seem to cause a loop just with _itself_.
